decidim-proposals/spec/commands/decidim/proposals/admin/import_proposals_spec.rb

```diff
@@ -134,6 +154,30 @@ module Admin
               end
             end
 
+            describe "proposal scopes" do
+              let(:states) { ProposalsImportForm::VALID_STATES.dup }
+              let(:scope) { create(:scope, organization: current_component.organization) }
+              let(:other_scope) { create(:scope, organization: current_component.organization) }
+
+              let(:scopes) { [scope] }
+              let(:scope_ids) { [scope.id] }
+
+              let!(:proposals) do
+                [
+                  create(:proposal, component: proposal.component, scope: scope),
+                  create(:proposal, component: proposal.component, scope: other_scope)
+                ]
+              end
+
+              it "only imports proposals from the selected scope" do
+                expect do
+                  command.call
+                end.to change { Proposal.where(component: current_component).count }.by(1)
+
+                expect(Proposal.where(component: current_component).scope.id).to eq(scope.id)
+              end
+            end
+
```

Comment:
@microstudi, any ideas on why the `ActionLog#save!` might be failing because of the scope?

```
Failures:

  1) Decidim::Proposals::Admin::ImportProposals call when the form is valid proposal scopes only imports proposals from the selected scope
     Failure/Error:
       Decidim.traceability.perform_action!(:create, Proposal, action_user, visibility: "all") do
         proposal = Proposal.new(attributes)
         proposal.add_coauthor(author, user_group: user_group_author)
         proposal.save!
         proposal
       end
     
     ActiveRecord::RecordInvalid:
       Validation failed: Scope is invalid
     # /home/vera/Decidim/decidim-core/app/services/decidim/action_logger.rb:47:in `log!'
     # /home/vera/Decidim/decidim-core/app/services/decidim/action_logger.rb:23:in `log'
     # /home/vera/Decidim/decidim-core/app/services/decidim/traceability.rb:122:in `log'
     # /home/vera/Decidim/decidim-core/app/services/decidim/traceability.rb:66:in `block (2 levels) in perform_action!'
     # /home/vera/Decidim/decidim-core/app/services/decidim/traceability.rb:63:in `block in perform_action!'
     # /home/vera/Decidim/decidim-core/app/services/decidim/traceability.rb:62:in `perform_action!'
     # ./app/services/decidim/proposals/proposal_builder.rb:18:in `create'
     # ./app/services/decidim/proposals/proposal_builder.rb:89:in `copy'
     # ./app/commands/decidim/proposals/admin/import_proposals.rb:36:in `block in import_proposals'
     # ./app/commands/decidim/proposals/admin/import_proposals.rb:33:in `map'
     # ./app/commands/decidim/proposals/admin/import_proposals.rb:33:in `import_proposals'
     # ./app/commands/decidim/proposals/admin/import_proposals.rb:25:in `call'
     # ./spec/commands/decidim/proposals/admin/import_proposals_spec.rb:175:in `block (6 levels) in <module:Admin>'
     # ./spec/commands/decidim/proposals/admin/import_proposals_spec.rb:173:in `block (5 levels) in <module:Admin>'

```

```
# decidim-core/app/services/decidim/action_logger.rb
  ...
  # Public: Logs the given `action` by the given `user` on the given
    # `resource`.
    #
    # Returns the newly created `Decidim::ActionLog` resource.
    def log!
      Decidim::ActionLog.create!(
        user: user,
        organization: organization,
        action: action,
        resource: resource,
        resource_id: resource.id,
        resource_type: resource.class.name,
        participatory_space: participatory_space,
        component: component,
        area: area,
        scope: scope,
        version_id: version_id,
        extra: extra_data,
        visibility: visibility
      )
    end


...

    def scope
      resource.try(:scope) || participatory_space.try(:scope)
    end
```

Reply:
I think ActionLogger uses `user.organization`, you should use a global organization object for all objects to fix it (it will also clarify a bit the code).